### DISCUSSION TITLE
Lock file compatibility

### DISCUSSION THREAD

**Discussion by liran-cohen-hs** [pdm-project/pdm#2164] [General]
Getting this error message when trying to verify lock file on PDM 2.7.4 that was created on PDM 2.8.2
"Lock file version is not compatible with PDM, installation may fail" and it resulted with error 1

is there a need/enforcement to work on the same version ? this means to coordinate the versions on developers computers and our CI/CD processes
I checked the changelog and didn't see any breaking changes so that's where my question comes from, trying to understand the approach here

Thanks

**Comment by frostming** [comment]
> is there a need/enforcement to work on the same version

Not the same version, but there is [a strategy](https://github.com/pdm-project/pdm/blob/fa34573ebf94a06dc42eb164908ef5b5ad435336/src/pdm/project/lockfile.py#L60-L72)[^1] to determine whether a lockfile is compatible. That is important, because the lockfile spec may evolve and at the same time we don't want to break things users are working on. It's good to have a way to know the lockfile spec has been upgraded and notify users to update it(or update it automatically)

[^1]: The strategy used to be more strict on PDM<2.8.0 and we have relaxed that.

**Reply by liran-cohen-hs** [reply]
followup on that - just bumped into another compatibility issue between 2.10.0 and 2.9.3 (lock created on the newer version)
I understand the need to break things sometimes, yet, in my opinion it should be mentioned in the changelog when it's happening

**Reply by frostming** [reply]
We usually don't care much about forward compatibility, and the lockfile version is also for backward compatibility, I hope for your understanding.

**Reply by liran-cohen-hs** [reply]
yes I understand
changed accordingly that our CI will always use latest version, so we shouldn't bump into problems (unless there's a breaking change :))
